Gems/Terrain/Code/Source/TerrainRenderer/TerrainMacroMaterialManager.cpp

```diff
@@ -323,12 +305,24 @@ namespace Terrain
         );
 
         macroMaterialShaderData.m_normalFactor = macroMaterial.m_data.m_normalFactor;
-        macroMaterialShaderData.m_boundsMin = { macroMaterial.m_data.m_bounds.GetMin().GetX(), macroMaterial.m_data.m_bounds.GetMin().GetY() };
-        macroMaterialShaderData.m_boundsMax = { macroMaterial.m_data.m_bounds.GetMax().GetX(), macroMaterial.m_data.m_bounds.GetMax().GetY() };
+        macroMaterial.m_data.m_minBounds.StoreToFloat2(macroMaterialShaderData.m_boundsMin.data());
+        macroMaterial.m_data.m_maxBounds.StoreToFloat2(macroMaterialShaderData.m_boundsMax.data());
         macroMaterialShaderData.m_colorMapId = macroMaterial.m_colorIndex;
         macroMaterialShaderData.m_normalMapId = macroMaterial.m_normalIndex;
     }
 
+    void TerrainMacroMaterialManager::AddMacroMaterialShaderEntry(uint16_t materialRef, MacroMaterialRefs& materialRefs)
+    {
+        for (auto& ref : materialRefs)
+        {
+            if (ref == InvalidMacroMaterialRef)
+            {
+                ref = materialRef;
+                break;
```

Comment:
Here can just return. Should it print some warning or error if all slots are already occupied?

Reply:
I don't think a warning is needed, it's just a limitation of the system. Once we add priorities the limitation will be less of an issue. Longer term we may want a different data partitioning scheme that allows for more materials in a given area.